packages/wdio-cucumber-framework/src/index.ts

```diff
@@ -483,6 +487,17 @@ class CucumberAdapter {
             const hookParams = getHookParams()
             const retryTest = isStep && isFinite(options.retry) ? options.retry : 0
 
+            let stepArg = hookParams?.step
+            if (!isStep) {
+                stepArg = {
+                    ...stepArg,
+                    name: options._wdioOptions?.name,
+                    tags: options._wdioOptions?.tags,
+                    hookType: options._wdioOptions?.hookType,
+                    body: options._wdioOptions?.body
+                }
+            }
```

Comment:
This seems to be more clear
```suggestion
            const stepArg = isStep
                ? hookParams?.step
                : {
                    name: options._wdioOptions?.name,
                    tags: options._wdioOptions?.tags,
                    hookType: options._wdioOptions?.hookType,
                    body: options._wdioOptions?.body
                }
```